contracts/DomainRegistry.sol

```diff
@@ -158,26 +182,38 @@ contract DomainRegistry is Initializable, OwnableUpgradeable {
     }
 
     /// @notice Allows an owner to withdraw their accumulated rewards
-    function withdraw() external {
-        uint256 share = _shares[msg.sender];
-        if (share == 0) {
-            revert WithdrawNoBalanceAvailable();
-        }
-        _shares[msg.sender] = 0;
-        (bool success, ) = msg.sender.call{value: share}("");
-        if (!success) {
-            revert DomainRegistryPaymentReverted({ controller: msg.sender });
+    function withdraw(string memory currency) external {
+        bool isEtn = keccak256(abi.encodePacked(currency)) == keccak256(abi.encodePacked('etn'));
+        if (isEtn) {
+            uint256 share = _shares[msg.sender];
+            if (share == 0) {
+                revert WithdrawNoBalanceAvailable();
+            }
+            _shares[msg.sender] = 0;
+            (bool success, ) = msg.sender.call{value: share}("");
+            if (!success) {
+                revert DomainRegistryPaymentReverted({ controller: msg.sender });
+            }
+        } else {
+            uint256 share = _tokens[msg.sender];
+            if (share == 0) {
+                revert WithdrawNoBalanceAvailable();
+            }
+            _tokens[msg.sender] = 0;
+            token.transfer(msg.sender, share);
```

Comment:
How can you withdraw tokens from contract address if you never send them to the contract address? There is no tokens sent during registration of domain.